nsqadmin/static/js/models/channel.js

```diff
@@ -12,6 +12,25 @@ var Channel = Backbone.Model.extend({
         return AppState.url('/topics/' +
             encodeURIComponent(this.get('topic')) + '/' +
             encodeURIComponent(this.get('name')));
+    },
+    initialize: function() {
+      this.on('change:clients', this.calculateHostnamePort);
+    },
+    calculateHostnamePort: function(m, v) {
+      
+      this.get("clients").forEach(function(v){
+        var client_id = v["client_id"]; 
+        var hostname = v["hostname"];
+        var remote_address = v["remote_address"];
+      
+        // ignore client_id if it's duplicative
+        if (client_id === hostname.split(".")[0] || client_id === hostname) {
+          v["client_id"] = "";
+        }
+
+        var port = remote_address.split(":").pop()
+        v['hostname_port'] =  hostname + ":" + port;
+      })
```

Comment:
I don't know enough about when the `clients` attribute would change and what values we might expect `client_id`, `hostname`, and `remote_address` to be to review the logic of this code. I might tweak the naming a bit to make what this method is doing a little more explicit. For example, maybe something like `setHostnamePort` is better?

Besides that, it's mostly just being consistent with style (e.g. indentation, single quotes, missing semicolons, spaces between `function()` and `{`).